### ISSUE TITLE
Check Format GitHub Action is failing

### ISSUE THREAD

**Issue by AObuchow** [devfile/devworkspace-operator#1314]
The GitHub Action's that [check the format](https://github.com/devfile/devworkspace-operator/blob/main/.github/workflows/pr.yml#L73-L81) of the repo's go code are currently failing with:
```
go: downloading golang.org/x/tools v0.25.0
go: golang.org/x/tools/cmd/goimports@latest (in golang.org/x/tools@v0.25.0): go.mod:3: invalid go version '1.22.0': must match format 1.23
```

I believe this is because we [use an outdated Go version](https://github.com/devfile/devworkspace-operator/blob/main/.github/workflows/pr.yml#L18) throughout the repo. I quickly tried to install goimports@latest in an "empty workspace" in Che:

```
$ go install golang.org/x/tools/cmd/goimports@latest
go: downloading golang.org/x/tools v0.25.0
go: golang.org/x/tools/cmd/goimports@latest: golang.org/x/tools@v0.25.0 requires go >= 1.22.0 (running go 1.21.11; GOTOOLCHAIN=local)
```

Originally, we kept this version of Go because Go 1.22 had [not yet been released for Fedora 40](https://github.com/devfile/devworkspace-operator/pull/1243), and I didn't want to break the local workflows of developers using Fedora.

Two solutions that come to mind:
- Pin go imports to v0.24.0 instead of latest
- Upgrade to go 1.22 throughout the repo, as it's supported in [our runner image](https://github.com/actions/runner-images/blob/main/images/ubuntu/Ubuntu2004-Readme.md#go).

**Comment by AObuchow** [comment]
@dkwon17 Since we're aiming to release DWO 0.31.0 today, what do you think about:
- Temporarily pin go imports to v0.24.0 instead of latest (to unblock pending PRs)
- After the upstream release, I can submit a PR to upgrade to go 1.22 throughout the repo and [unpin set-envtest](https://github.com/devfile/devworkspace-operator/pull/1243). This is a bit more of a delicate process, and I don't want it to block the DWO release.

**Comment by AObuchow** [comment]
Re-opening this issue so we can remember to proceed with the more permanent solution of upgrading the repo to go 1.22 throughout the repo.

**Comment by AObuchow** [comment]
> Re-opening this issue so we can remember to proceed with the more permanent solution of upgrading the repo to go 1.22 throughout the repo.

Never mind, I decided to make a separate issue for this: https://github.com/devfile/devworkspace-operator/issues/1316
